components/RegionDialog/RegionDialog.tsx

```diff
@@ -54,56 +55,73 @@ export function RegionDialog({ isOpen, onClose }: RegionDialogProps) {
   }
 
   return (
-    <div
-      className="min-w-screen h-screen fixed  left-0 top-0 flex justify-center items-center inset-0 z-50 outline-none focus:outline-none bg-no-repeat bg-center"
-      id="modal-id"
-    >
-      <div className="absolute bg-black opacity-50 inset-0 z-0" />
-      <div className="w-full  max-w-lg pb-6 relative mx-auto my-auto rounded-sm shadow-lg flex flex-col items-center bg-white ">
-        <div className="text-center p-5 mt-4 flex-auto justify-center">
-          <h2 className="text-2xl font-bold py-4">{t.formatMessage(messages.regionModalHeader)}</h2>
-        </div>
-        <div className="text-gray-300 py-1 mb-4 flex flex-col items-center w-3/5">
-          <p className="text-sm text-gray-500 self-start mb-1">
-            {t.formatMessage(messages.channelFieldLabel)}
-          </p>
-          <select
-            className="w-full text-black"
-            id="channel"
-            {...register("channel", {
-              required: true,
-            })}
-          >
-            {channels.map(({ slug, name }) => (
-              <option key={slug} value={slug}>
-                {name}
-              </option>
-            ))}
-          </select>
-        </div>
-        <div className="text-gray-300 py-1 pb-3 flex flex-col items-center w-3/5">
-          <p className="text-sm text-gray-500 self-start mb-1">
-            {t.formatMessage(messages.languageFieldLabel)}
-          </p>
-          <select
-            className="w-full text-black"
-            id="locale"
-            {...register("locale", {
-              required: true,
-            })}
-          >
-            {LOCALES.map((locale) => (
-              <option key={locale.slug} value={locale.slug}>
-                {locale.name}
-              </option>
-            ))}
-          </select>
-        </div>
-        <div className="p-3  mt-4 mb-4 text-center space-x-4 md:block">
-          <Button onClick={onSubmit}>{t.formatMessage(messages.saveButton)}</Button>
+    <Dialog as="div" className="fixed inset-0 z-10 overflow-y-auto" open={isOpen} onClose={onClose}>
+      <div className="min-h-screen px-4 text-center">
+        <Dialog.Overlay className="fixed inset-0 bg-black opacity-50" />
+
+        {/* This element is to trick the browser into centering the modal contents. */}
```

Comment:
It looks like this can be achieved with just pure css?

Reply:
Yup. But decided that don't want to introduce another approach before the redesign and went with the pure tailwind.

Reply:
By pure css I mean using tailwind and not using html tricks to center contents. This should be easily done with flex

Reply:
You are totally right (alignment pun intended)